### PR TITLE
Provider last fix

### PR DESCRIPTION
**Pull request by bekeles**

### CODE REVIEW

**Comment by depryf** [comment]
@bekeles the check failed because of code duplication in your new additions. Can you confirm it was like that already and you didn't really make anything worse with your changes?

File: src/main/java/com/imsweb/mph/MphUtils.java [hunk 1]
```diff
@@ -72,55 +72,34 @@ public enum MpResult {
         INVALID_INPUT
     }
 
-    // the unique instance of this utility class
-    private static MphUtils _INSTANCE = null;
```
Comment (depryf) [inline src/main/java/com/imsweb/mph/MphUtils.java:76]: I am confused by this.   You are removing the instance? Are you expecting every software to now cache they own instance? Except for SEER*API, no software need to support a provider.   Right now SEER*Abs does something like "MphUtils.getInstance().xxx()".   Are you expecting that software to have to do "new MphUtils().xxx()" and create a new instance every time? Or to cache it itself? Neither of those options look good.  Why did we have to remove the instance?

**Comment by bekeles** [comment]
No I didn't make anything worse. The after merge duplication rate is still at 12.5%

File: src/main/java/com/imsweb/mph/HematoUtils.java [hunk 2]
```diff
@@ -9,37 +9,34 @@ public final class HematoUtils {
 
     private static Pattern _MORPHOLOGY = Pattern.compile("^(\\d{4}/\\d)");
 
-    public static boolean isSamePrimary(String morph1, String morph2, int year1, int year2) {
```
Comment (depryf) [inline src/main/java/com/imsweb/mph/HematoUtils.java:12]: I really don't love these changes.   You are not requiring everyone to provide a provider to each of these methods. 

But except for SEER*API, no software cares or use a provider, they all use the default one. So now they will need to get the default provider for these calls? But not to use MphUtils itself? That seems all wrong.

I would remove this utility class and move these methods to MphUtils, and make them non-static and use the provider from that class. 

Why having multiple entry points for such a simple library; and if these methods need a provider, putting them in MphUtils, which has a provider already, makes much more sense IMO.

File: src/main/java/com/imsweb/mph/mpgroups/Mp2010HematopoieticGroup.java [hunk 3]
```diff
@@ -321,7 +322,7 @@ public TempRuleResult apply(MphInput i1, MphInput i2) {
                         result.setPotentialResult(MphUtils.MpResult.MULTIPLE_PRIMARIES);
                         result.setMessageUnknownDiagnosisDate(this.getStep(), this.getGroupName());
                     }
-                    else if (MphConstants.DATE_VERIFY_APART == daysApart && latestDx > 0 && HematoUtils.isChronicToAcuteTransformation(MphConstants.COMPARE_DX_FIRST_LATEST == latestDx ? morph2 : morph1,
+                    else if (MphConstants.DATE_VERIFY_APART == daysApart && latestDx > 0 && HematoUtils.isChronicToAcuteTransformation(provider, MphConstants.COMPARE_DX_FIRST_LATEST == latestDx ? morph2 : morph1,
```
Comment (depryf) [inline src/main/java/com/imsweb/mph/mpgroups/Mp2010HematopoieticGroup.java:325]: If you feel strongly about keeping this solution, we can go with that.

But personally, I would have created a "RuleExecutionContext" class that takes an instance of "MphUtils" in its constructor (and caches it).  I would have moved the methods from HematoDbUtils to MphUtils, and I would do something like "context.getMphUtils().isChronicToAcuteTransformation()" in this method.

Or you could also add a "isChronicToAcuteTransformation()" convenience method to the context so you skip the "getMphUtils" call...

Personally I think this is a cleaner solution than having rules that call a static method. This fill all weirdly connected.

**Comment by depryf** [comment]
I like these changes much better! Merging now.

**Pull request merged**